src/format.iced

```diff
@@ -0,0 +1,122 @@
+stream = require('keybase-chunk-stream')
+util = require('./util')
+
+space = Buffer.from(' ')
+newline = Buffer.from('\n')
+punctuation = Buffer.from('.')
+
+words_per_line = 200
+chars_per_word = 15
+
+noop = () ->
+
+exports.FormatStream = class FormatStream extends stream.ChunkStream
+
+  __format : (chunk) ->
+    return Buffer.concat([space, chunk])
+
+  _format : (chunk) ->
+    if chunk.length < chars_per_word then return chunk
+    results = []
+    for i in [0...chunk.length] by chars_per_word
+      word = chunk[i...i+chars_per_word]
+      if i+chars_per_word >= chunk.length
+        results.push(word)
+      else
+        if @_word_count % 200 is 0 and @_word_count isnt 0 then word = Buffer.concat([word, newline])
+        else word = Buffer.concat([word, space])
+        ++@_word_count
+        results.push(word)
+    return Buffer.concat(results)
+
+  _transform : (chunk, encoding, cb) ->
+    if not @_header_written
+      @push(Buffer.concat([@_header, punctuation, space]))
+      @_header_written = true
+    super(chunk, encoding, cb)
+
+  _flush : (cb) ->
+    super(noop)
+    @push(Buffer.concat([punctuation, space, @_footer, punctuation]))
+    cb()
+
+  constructor : (opts) ->
+    if opts?.brand? then _brand = opts.brand else _brand = 'KEYBASE'
+    @_header = Buffer.from("BEGIN#{space}#{_brand}#{space}SALTPACK#{space}ENCRYPTED#{space}MESSAGE")
+    @_footer = Buffer.from("END#{space}#{_brand}#{space}SALTPACK#{space}ENCRYPTED#{space}MESSAGE")
+    @_header_written = false
+    @_word_count = 0
+    super(@_format, {block_size : 1, exact_chunking : false, writableObjectMode : false, readableObjectMode : false})
+
+exports.DeformatStream = class DeformatStream extends stream.ChunkStream
+
+  _header_mode = 0
+  _body_mode = 1
+  _footer_mode = 2
+  _strip_chars = Buffer.from('>\n\r\t ')
+
+  _strip = (chunk) ->
+    indicies = []
+    ret = []
+    for i in [0...chunk.length]
+      if _strip_chars.indexOf(chunk[i]) is -1 then ret.push(chunk[i])
+    return Buffer.from(ret)
+
+  _deformat : (chunk) ->
+   if @_mode is _header_mode
+      index = chunk.indexOf(punctuation[0])
+      if index isnt -1
+        # we found the period
+        read_header = chunk[0...index]
+        read_header = _strip(read_header)
+        unless util.bufeq_secure(read_header, _strip(@_header)) then throw new Error("Header failed to verify!")
+        @_mode = _body_mode
+        @block_size = 1
+        @exact_chunking = false
+        @extra = chunk[index+punctuation.length+space.length...]
+        # so that we can't enter this if statement more than once
+        _header_mode = null
+        return Buffer.from('')
+      else
+        # something horrible happened
+        throw new Error('Somehow didn\'t get a full header packet')
```

Comment:
It looks like you have an exact chunking length equal to the expected header. But if I insert _multiple_ spaces between words (or leading spaces/newlines), the header length will be longer, and it won't fit inside an entire chunk. I think we need to buffer chunks until we see that period, rather than assuming the first chunk will include the whole header. We could strip excess whitespace on the way in, to avoid unbounded memory consumption in the case of a pathological number of spaces, but I think we should support it somehow.